pincer/commands/arg_types.py

```diff
@@ -0,0 +1,226 @@
+# Copyright Pincer 2021-Present
+# Full MIT License can be found in `LICENSE` at the project root.
+
+from typing import Any, List, Tuple, Union, T
+
+from ..utils.types import MISSING
+from ..objects.app.command import AppCommandOptionChoice
+
+
+class _CommandTypeMeta(type):
+    def __getitem__(cls, args: Union[Tuple, Any]):
+        if not isinstance(args, tuple):
+            args = (args,)
+
+        return cls(*args)
+
+
+class CommandArg(metaclass=_CommandTypeMeta):
+    """
+    Holds the parameters of an application command option
+
+    .. code-block:: python3
+
+        CommandArg[
+            # This is the type of command.
+            # Supported types are str, int, bool, float, User, Channel, and Role
+            int,
+            # The modifiers to the command go here
+            Description["Pick a number 1-10"],
+            MinValue[1],
+            MaxValue[10]
+        ]
+
+    Parameters
+    ----------
+    command_type : T
+        The type of the command
+    \*args : :class:`~pincer.commands.arg_types.Modifier`
```

Comment:
@Endercheif Is this supposed to be one slash or two slashes like `\\*\\*kwargs` in other places?